File: patch_list.py
with open("components/inbox/conversation-list.tsx", "r") as f:
    text = f.read()

text = text.replace(
'''      {/* Search */}
      <div className="p-4 shrink-0 relative z-10">
        <div className="relative group">
          <Search className="absolute left-4 top-1/2 h-4 w-4 -translate-y-1/2 text-[var(--ink-3)] group-focus-within:text-[var(--brand)] transition-colors" />
          <input
            type="text"
            placeholder="Search conversations..."
            value={filters.search}
            onChange={(e) => setFilters({ search: e.target.value })}
            className="w-full rounded-2xl border border-[var(--border)] bg-[var(--surface)]/80 backdrop-blur-md py-3 pl-11 pr-10 text-[13px] font-medium placeholder:text-[var(--ink-3)] focus:outline-none focus:ring-2 focus:ring-[var(--brand)] focus:border-transparent transition-all shadow-sm"
          />
          {filters.search && (
            <button
              onClick={() => setFilters({ search: "" })}
              className="absolute right-3 top-1/2 -translate-y-1/2 text-[var(--ink-2)] hover:text-[var(--ink)] hover:bg-[var(--surface-2)] p-1.5 rounded-full transition-colors"
            >
              <X className="h-3.5 w-3.5" />
            </button>
          )}
        </div>
      </div>

      <div className="flex flex-col gap-3 px-4 pb-4 shrink-0 relative z-10 border-b border-[var(--border)]/40">
        {/* Status Filters */}
        <div className="flex items-center gap-1.5 overflow-x-auto scrollbar-none [scrollbar-width:none] [-ms-overflow-style:none] [&::-webkit-scrollbar]:hidden min-w-0 pb-1">
          {["all", "open", "closed", "snoozed", "archived"].map((status) => (
            <button
              key={status}
              onClick={() => setFilters({ status: status as any })}
              className={cn(
                "rounded-full px-4 py-1.5 text-[11px] font-bold capitalize transition-all shrink-0 border",
                filters.status === status
                  ? "bg-[var(--surface)] text-[var(--ink)] shadow-md border-[var(--border)]"
                  : "bg-transparent border-transparent text-[var(--ink-2)] hover:bg-[var(--surface)] hover:text-[var(--ink)]",
              )}
            >
              {status}
            </button>
          ))}
        </div>

        {/* Channel Dropdown */}
        <div className="flex items-center gap-2 shrink-0 bg-[var(--surface)]/50 backdrop-blur-sm rounded-xl p-1.5 border border-[var(--border)]/50 w-max">
          <select
            value={filters.channelId}
            onChange={(e) => setFilters({ channelId: e.target.value })}
            className="text-[11px] font-bold bg-transparent px-3 py-1 focus:outline-none text-[var(--ink)] max-w-[140px] truncate shrink-0 cursor-pointer appearance-none"
            title="Filter by connected page or account"
          >
            <option value="all">
              {filters.platform !== "all"
                ? `All ${filters.platform.toUpperCase()}`
                : "All Channels"}
            </option>
            {channelsForActivePlatform.map((ch) => (
              <option key={ch.id} value={ch.id}>
                {ch.name}
              </option>
            ))}
          </select>
          {filters.channelId !== "all" && (
            <button
              onClick={() => setFilters({ channelId: "all" })}
              className="p-1 rounded-full text-[var(--ink-2)] hover:text-[var(--danger)] hover:bg-[var(--danger)]/10 transition-colors"
              title="Reset channel filter"
            >
              <X className="h-3.5 w-3.5" />
            </button>
          )}
        </div>
      </div>''',
'''      {/* Search & Status Filters */}
      <div className="flex flex-col gap-2 p-3 shrink-0 relative z-10 border-b border-[var(--border)]/40 bg-[var(--surface-2)]/30">
        <div className="relative group">
          <Search className="absolute left-3 top-1/2 h-3.5 w-3.5 -translate-y-1/2 text-[var(--ink-3)] group-focus-within:text-[var(--brand)] transition-colors" />
          <input
            type="text"
            placeholder="Search conversations..."
            value={filters.search}
            onChange={(e) => setFilters({ search: e.target.value })}
            className="w-full rounded-xl border border-[var(--border)] bg-[var(--surface)] py-2 pl-9 pr-8 text-xs font-medium placeholder:text-[var(--ink-3)] focus:outline-none focus:ring-1 focus:ring-[var(--brand)] focus:border-transparent transition-all shadow-sm"
          />
          {filters.search && (
            <button
              onClick={() => setFilters({ search: "" })}
              className="absolute right-2 top-1/2 -translate-y-1/2 text-[var(--ink-2)] hover:text-[var(--ink)] hover:bg-[var(--surface-2)] p-1 rounded-full transition-colors"
            >
              <X className="h-3 w-3" />
            </button>
          )}
        </div>

        <div className="flex items-center gap-1.5 overflow-x-auto scrollbar-none [scrollbar-width:none] [-ms-overflow-style:none] [&::-webkit-scrollbar]:hidden min-w-0">
          {["all", "open", "closed", "snoozed", "archived"].map((status) => (
            <button
              key={status}
              onClick={() => setFilters({ status: status as any })}
              className={cn(
                "rounded-lg px-3 py-1.5 text-[10.5px] font-bold capitalize transition-all shrink-0 border",
                filters.status === status
                  ? "bg-[var(--surface)] text-[var(--ink)] shadow-sm border-[var(--border)]"
                  : "bg-transparent border-transparent text-[var(--ink-2)] hover:bg-[var(--surface)] hover:text-[var(--ink)]",
              )}
            >
              {status}
            </button>
          ))}
        </div>
      </div>'''
)

text = text.replace(
'''      {/* Conversation list */}
      <div className="flex-1 overflow-y-auto divide-y divide-border/60 flex flex-col">''',
'''      {/* Conversation list */}
      <div className="flex-1 overflow-y-auto divide-y divide-[var(--border)]/30 flex flex-col pb-[72px] md:pb-0">'''
)

with open("components/inbox/conversation-list.tsx", "w") as f:
    f.write(text)

File: patch_list.js
const fs = require('fs');
let content = fs.readFileSync('components/inbox/conversation-list.tsx', 'utf8');

// Replace the layout styling for the container and filters
const newHeader = `    <div className="flex flex-col h-full bg-[var(--surface-2)]">
      {/* Search Header */}
      <div className="p-3 pb-2 shrink-0">
        <div className="relative group">
          <Search className="absolute left-3 top-2.5 h-4 w-4 text-[var(--ink-3)] group-focus-within:text-[var(--brand)] transition-colors" />
          <input
            type="text"
            placeholder="Search conversations..."
            value={searchQuery}
            onChange={(e) => setSearchQuery(e.target.value)}
            className="w-full bg-[var(--surface)] border border-[var(--border)] pl-9 pr-3 py-2 text-[13px] rounded-lg shadow-sm focus:outline-none focus:ring-1 focus:ring-[var(--brand)] focus:border-[var(--brand)] transition-all placeholder:text-[var(--ink-3)]"
          />
        </div>
      </div>

      {/* Filters */}
      <div className="px-3 py-2 flex items-center justify-between shrink-0">
        <div className="flex items-center gap-1 bg-[var(--surface)] p-1 rounded-lg border border-[var(--border)]/50 w-full">
          <FilterButton active={filter === "all"} onClick={() => setFilter("all")} label="All" />
          <FilterButton active={filter === "open"} onClick={() => setFilter("open")} label="Open" />
          <FilterButton active={filter === "closed"} onClick={() => setFilter("closed")} label="Closed" />
          <FilterButton active={filter === "snoozed"} onClick={() => setFilter("snoozed")} label="Snoozed" />
          <FilterButton active={filter === "archived"} onClick={() => setFilter("archived")} label="Archived" />
        </div>
      </div>

      {/* Channel Filters */}
      <div className="px-3 py-2 flex items-center gap-1.5 overflow-x-auto no-scrollbar shrink-0 border-b border-[var(--border)]/50 bg-[var(--surface-2)]">
        <button
          onClick={() => setSelectedChannel("all")}
          className={cn(
            "flex items-center gap-1.5 px-3 py-1.5 rounded-md text-[13px] font-medium whitespace-nowrap transition-all duration-200 border",
            selectedChannel === "all"
              ? "bg-[var(--surface)] border-[var(--border)] text-[var(--ink)] shadow-sm"
              : "bg-transparent border-transparent text-[var(--ink-3)] hover:text-[var(--ink-2)] hover:bg-[var(--surface)]/50"
          )}
        >
          All Channels
        </button>
        {channels.map((channel) => (
          <button
            key={channel.id}
            onClick={() => setSelectedChannel(channel.id)}
            className={cn(
              "flex items-center gap-1.5 px-3 py-1.5 rounded-md text-[13px] font-medium whitespace-nowrap transition-all duration-200 border",
              selectedChannel === channel.id
                ? "bg-[var(--surface)] border-[var(--border)] text-[var(--ink)] shadow-sm"
                : "bg-transparent border-transparent text-[var(--ink-3)] hover:text-[var(--ink-2)] hover:bg-[var(--surface)]/50"
            )}
          >
            {channel.display_name || channel.platform}
          </button>
        ))}
      </div>

      {/* List */}
      <div className="flex-1 overflow-y-auto overflow-x-hidden p-2 space-y-0.5">
        <AnimatePresence initial={false}>
          {filtered.length === 0 ? (
            <motion.div
              initial={{ opacity: 0 }}
              animate={{ opacity: 1 }}
              className="p-8 text-center"
            >
              <div className="mx-auto w-12 h-12 rounded-full bg-[var(--surface)] flex items-center justify-center mb-4">
                <Search className="h-5 w-5 text-[var(--ink-3)]" />
              </div>
              <p className="text-[13px] font-medium text-[var(--ink-2)]">
                No conversations found.
              </p>
            </motion.div>
          ) : (
            filtered.map((conv) => (
              <motion.button
                layout="position"
                initial={{ opacity: 0, y: 10 }}
                animate={{ opacity: 1, y: 0 }}
                exit={{ opacity: 0, scale: 0.95 }}
                key={conv.id}
                onClick={() => onSelect(conv)}
                className={cn(
                  "w-full text-left p-3 rounded-lg flex gap-3 transition-all duration-200 relative group items-start border border-transparent",
                  selectedId === conv.id
                    ? "bg-[var(--paper)] shadow-sm border border-[var(--border)] ring-1 ring-black/5 z-10"
                    : "hover:bg-[var(--surface)]/60 hover:border-[var(--border)]/50"
                )}
              >
                {/* Active Indicator */}
                {selectedId === conv.id && (
                  <div className="absolute left-0 top-1/2 -translate-y-1/2 w-1 h-8 bg-[var(--brand)] rounded-r-full" />
                )}`;

const listStart = content.indexOf('    <div className="flex flex-col h-full bg-transparent">');
const listEnd = content.indexOf('                {/* Active Glow Indicator */}');
const activeGlowIndicatorEnd = content.indexOf('                <div className="relative shrink-0 mt-1">');

const replaced = content.substring(0, listStart) + newHeader + '\n\n                <div className="relative shrink-0 mt-0.5">' + content.substring(activeGlowIndicatorEnd + 72);

// Fix FilterButton
const newFilterButton = `function FilterButton({
  active,
  onClick,
  label,
}: {
  active: boolean;
  onClick: () => void;
  label: string;
}) {
  return (
    <button
      onClick={onClick}
      className={cn(
        "flex-1 py-1 px-2 rounded-md text-[13px] font-medium transition-all duration-200 text-center",
        active
          ? "bg-[var(--paper)] text-[var(--ink)] shadow-sm border border-[var(--border)]/50"
          : "text-[var(--ink-3)] hover:text-[var(--ink)] hover:bg-[var(--surface-2)]"
      )}
    >
      {label}
    </button>
  );
}`;
const buttonStart = replaced.indexOf('function FilterButton({');
const buttonEnd = replaced.indexOf('}', replaced.indexOf('</button>')) + 1;
const finalContent = replaced.substring(0, buttonStart) + newFilterButton + replaced.substring(buttonEnd);

fs.writeFileSync('components/inbox/conversation-list.tsx', finalContent);
